src/controllers/controlpickermenu.cpp

```diff
@@ -415,6 +415,57 @@ ControlPickerMenu::ControlPickerMenu(QWidget* pParent)
         hotcueMainMenu->addMenu(hotcueMoreMenu);
     }
 
+    // Intro/outro range markers
+    QMenu* introOutroMenu = addSubmenu(tr("Intro / Outro Cues"));
+    QString introStartTitle(tr("Intro Start Cue"));
+    QString introEndTitle(tr("Intro End Cue"));
+    QString outroStartTitle(tr("Outro Start Cue"));
+    QString outroEndTitle(tr("Outro End Cue"));
+    QMenu* introStartMenu = addSubmenu(introStartTitle, introOutroMenu);
+    QMenu* introEndMenu = addSubmenu(introEndTitle, introOutroMenu);
+    QMenu* outroStartMenu = addSubmenu(outroStartTitle, introOutroMenu);
+    QMenu* outroEndMenu = addSubmenu(outroEndTitle, introOutroMenu);
```

Comment:
We can save a whopping 4 lines if we move these into the initializer list as well :D

Reply:
"It's a kinda magic. Magic, magic..."
It even works to create each menu in the for loop, but I guess that's to much and I can't tell if that's good pratcice.
but it would be pretty compact then..

Reply:
Funny thing is if the menus are created like that `introStartMenu` is listed last in the controls menu
``` c++
    QMenu* introEndMenu = addSubmenu(introEndTitle, introOutroMenu);
    QMenu* outroStartMenu = addSubmenu(outroStartTitle, introOutroMenu);
    QMenu* outroEndMenu = addSubmenu(outroEndTitle, introOutroMenu);
    QList<QMenu*> introOutroSubmenus = {
            addSubmenu(introStartTitle, introOutroMenu), // introStartMenu
            introEndMenu,
            outroStartMenu,
            outroEndMenu};
```
so the elements' order in initializer doesn't necessarily reflect the actual order.

Reply:
I'd like to do it like this now, let me know if that's okay:
``` c++
    // Intro/outro range markers
    QMenu* introOutroMenu = addSubmenu(tr("Intro / Outro Cues"));
    const QStringList cueTypeTitles = {
            tr("Intro Start Cue"),
            tr("Intro End Cue"),
            tr("Outro Start Cue"),
            tr("Outro End Cue")};
    const QStringList cueTypeNames = {
            tr("intro start cue"),
            tr("intro end cue"),
            tr("outro start cue"),
            tr("outro end cue")};
    const QStringList cueTypeCOs = {
            "intro_start",
            "intro_end",
            "outro_start",
            "outro_end"};

    for (int i = 0; i < cueTypeTitles.size(); ++i) {
        QMenu* tempMenu = addSubmenu(cueTypeTitles[i], introOutroMenu);
        addDeckAndSamplerAndPreviewDeckControl(
                QString("%1_activate").arg(cueTypeCOs[i]),
                tr("Activate %1").arg(cueTypeTitles[i]),
                tr("Jump to or set the %1").arg(cueTypeNames[i]),
                tempMenu);
        addDeckAndSamplerAndPreviewDeckControl(
                QString("%1_set").arg(cueTypeCOs[i]),
                tr("Set %1").arg(cueTypeTitles[i]),
                tr("Set or jump to the %1").arg(cueTypeNames[i]),
                tempMenu);
        addDeckAndSamplerAndPreviewDeckControl(
                QString("%1_clear").arg(cueTypeCOs[i]),
                tr("Clear %1").arg(cueTypeTitles[i]),
                tr("Clear the %1").arg(cueTypeNames[i]),
                tempMenu);
    }
```

Reply:
Yup, looks fine.